pkg/registry/resource/resourceclaim/strategy_test.go

```diff
@@ -141,22 +188,34 @@ const (
 )
 
 func TestStrategy(t *testing.T) {
-	if !Strategy.NamespaceScoped() {
+	fakeClient := fake.NewSimpleClientset()
+	mockNSClient := fakeClient.CoreV1().Namespaces()
+	strategy := NewStrategy(legacyscheme.Scheme, names.SimpleNameGenerator, mockNSClient)
+	if !strategy.NamespaceScoped() {
 		t.Errorf("ResourceClaim must be namespace scoped")
 	}
-	if Strategy.AllowCreateOnUpdate() {
+	if strategy.AllowCreateOnUpdate() {
 		t.Errorf("ResourceClaim should not allow create on update")
 	}
+	if strategy.nsClient == nil {
+		t.Errorf("ResourceClaim nsClient should not be nil")
+	}
 }
 
 func TestStrategyCreate(t *testing.T) {
 	ctx := genericapirequest.NewDefaultContext()
-
+	fakeClient := fake.NewSimpleClientset()
+	mockNSClient := fakeClient.CoreV1().Namespaces()
+	err := CreateNamespace(mockNSClient)
+	if err != nil {
+		t.Fatalf("failed to create test namespaces: %v", err)
+	}
```

Comment:
Normally you just pass in the objects you need to exist into the fake client.

```suggestion
	fakeClient := fake.NewSimpleClientset(ns1, ns2)
	mockNSClient := fakeClient.CoreV1().Namespaces()
```